spectrometer.py

```diff
@@ -1,303 +1,331 @@
-import os,sys, math, time
-from PIL import Image, ImageDraw, ImageFile, ImageFont
+import sys
+import math
+import time
 import picamera
-from collections import OrderedDict
 from fractions import Fraction
+from collections import OrderedDict
+from PIL import Image, ImageDraw, ImageFile, ImageFont
+
+
+# scan a column to determine top and bottom of area of lightness
+def get_spectrum_y_bound(pix, x, middle_y, spectrum_threshold, spectrum_threshold_duration):
+    c = 0
+    spectrum_top = middle_y
+    for y in range(middle_y, 0, -1):
+        r, g, b = pix[x, y]
+        brightness = r + g + b
+        if brightness < spectrum_threshold:
+            c = c + 1
+            if c > spectrum_threshold_duration:
+                break
+        else:
+            spectrum_top = y
+            c = 0
+
+    c = 0
+    spectrum_bottom = middle_y
+    for y in range(middle_y, middle_y * 2, 1):
+        r, g, b = pix[x, y]
+        brightness = r + g + b
+        if brightness < spectrum_threshold:
+            c = c + 1
+            if c > spectrum_threshold_duration:
+                break
+        else:
+            spectrum_bottom = y
+            c = 0
+
+    return spectrum_top, spectrum_bottom
+
+
+# find aperture on right hand side of image along middle line
+def find_aperture(pic_pixels, pic_width: int, pic_height: int) -> object:
+    middle_x = int(pic_width / 2)
```

Comment:
Slightly more pythonic: `pic_width // 2`

For example:
```
In [14]: int(3 / 2)
Out[14]: 1

In [15]: 3 // 2
Out[15]: 1
```

Reply:
Note, the above is from ipython, which I highly recommend installing if you haven't yet.

Reply:
Ohhh NICE! I noticed that the interpreter complained that these were floats after the Python 3 jump. That's a much more elegant fix. Thanks!